File: SecondLargest.cpp
// Optimal solution for Second largest 

#include <iostream>
#include <vector>
using namespace std;
// In global space of integer array is 10^7
//global array
vector<int> arr= {1,2,6,8,5};

int LargestandSecondLargest(vector<int> arr){
    int largest = arr[0];
    int secondLargest = -1;

for(int i = 0; i< arr.size(); i++){
    if(arr[i] > largest){
        secondLargest = largest; // previous largest moves to second largest
        largest = arr[i];
    }
    else if(arr[i] < largest && arr[i] > secondLargest){ 
        secondLargest = arr[i];
    } 
   }
  return secondLargest;
}
int main(){
    
    //In main function the integer array size is 10^6

    int secondLargest = LargestandSecondLargest(arr);
    cout<<"SecondLargest Elem: "<<secondLargest;
}

File: secondLargest.py
# Find the Second Largest Element from the Array

a = [1,2,4,7,7,5]

largest = 0
secondLargest = 0 
for i in a:
    if(i>largest):
        largest = i
        
for i in a:
    if(i>secondLargest and i!=largest):
        secondLargest = i

print("Second largest Element in Array :",secondLargest) 